src/Servers/Kestrel/shared/HPackHeaderWriter.cs

```diff
@@ -10,14 +10,26 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2;
 namespace Microsoft.AspNetCore.Server.Kestrel.Core.Tests;
 #endif
 
+internal enum HeaderWriteResult : byte
```

Comment:
if this is entirely contextual to `HPackHeaderWriter`, I wonder if this should be nested; it would also avoid the temptation to yell "new file"